server/app/routes.py

```diff
@@ -13,3 +20,44 @@ def videos_raw():
 	@app.route('/videos')
 	def videos():
 		return jsonify(rss.summarize_watchlist(app.config['watchlist']))
+
+	@app.route('/watchlist', methods=['POST', 'DELETE'])
+	async def modify_watchlist():
+		# only required value is 'id'
+		json = request.get_json()
+		if 'id' not in json and 'url' not in json:
+			return '', 400 # bad request
+		if 'id' in json and 'url' in json:
+			return '', 400 # bad request
+
+		args = {}
+		if 'id' in json:
+			_copy_values(args, json, ('id', 'source', 'site', 'name'))
+		if 'url' in json:
+			id = await url_to_video_id(json['url'])
+			if id is None: #pragma: nocover
+				# channel id is not found
+				return '', 404 # Not Found
```

Comment:
i think this is more of a 400 error (but I don't feel strongly) and a message would be nice "couldn't extract channel id from url"

Reply:
There are multiple errors that could lead to id being None.
1. Url is incorrect (400)
2. Url is correct but the channel name does not correspond to any channel (404)

I went with 404 since it felt more correct compared to the other parts where I return 400.

Reply:
Added error message